### DISCUSSION TITLE
Why does @envelop/sentry v3 skip `GraphQLError`

### DISCUSSION THREAD

**Discussion by morishin** [dotansimha/envelop#1569] [Q&A] [answered]
https://www.the-guild.dev/graphql/envelop/v3/guides/migrating-from-v2-to-v3#1envelopsentry-default-will-skip-all-graphqlerror says _@envelop/sentry default will skip all GraphQLError_.

I was wondering why @envelop/sentry skips `GraphQLError`.
I think the developers should notice any error occured in GraphQL server app. For example, It should send an error to Sentry when GraphQL resolvers return null value for non-null fields.

Please let me know the motivation if you know. 🙏 

https://github.com/n1ru4l/envelop/blob/3a11aff3b3ac83431bcc70a78ab8104a5ed0bee3/packages/plugins/sentry/src/index.ts#L88-L93

**Comment by morishin** [comment]
By the way, this description seems outdated because it includes `EnvelopError` dropped from v3 💭 
https://github.com/n1ru4l/envelop/blob/3a11aff3b3ac83431bcc70a78ab8104a5ed0bee3/packages/plugins/sentry/README.md?plain=1#L69

**Comment by morishin** [comment]
I guess the reason is that:
1. Every exceptions will be wrapped with `GraphQLError` by GraphQL server.
1. The only case when `originalError` is `GraphQLError` is that a developer raise `GraphQLError` with writing `new GraphQLError(...)` intentionally.
1. Therefore `GraphQLError` should not be sent to Sentry and [not be masked](https://github.com/n1ru4l/envelop/issues/1343).

**Comment by n1ru4l** [comment, marked as answer]
Hey @morishin, sorry for the late reply here.

This has been a bug in v3 and will be resolved with v4. See the PR https://github.com/n1ru4l/envelop/pull/1583

We added tests to make sure this regression will not happen again.
